frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/upload-pvc-popover.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from 'react';
+import { StatusIconAndText } from '@console/shared';
+import { InProgressIcon, ErrorCircleOIcon } from '@patternfly/react-icons';
+import {
+  Popover,
+  PopoverPosition,
+  Button,
+  Progress,
+  ProgressVariant,
+  Stack,
+  StackItem,
+  Spinner,
+} from '@patternfly/react-core';
+import { UPLOAD_STATUS } from './consts';
+import { DataUpload, CDIUploadContext } from './cdi-upload-provider';
+import { killUploadPVC } from '../../k8s/requests/cdi-upload/cdi-upload-requests';
+
+export const getProgressVariant = (status: string) => {
+  switch (status) {
+    case UPLOAD_STATUS.ERROR:
+      return ProgressVariant.danger;
+    case UPLOAD_STATUS.SUCCESS:
+      return ProgressVariant.success;
+    default:
+      return null;
+  }
+};
+
+export const UploadPVCPopover: React.FC<PVCUploadStatusProps> = ({ pvcName, namespace }) => {
+  const { uploads } = React.useContext(CDIUploadContext);
+  const upload: DataUpload = uploads.find(
```

Comment:
no need to specify type as `uploads` is typed properly
```suggestion
  const upload = uploads.find(
```